hazelcast/src/main/java/com/hazelcast/wan/WanEventCounter.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.wan;
+
+import com.hazelcast.util.ConstructorFunction;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.hazelcast.util.ConcurrencyUtil.getOrPutIfAbsent;
+
+/**
+ * Counter for WAN events for a single distributed object type (map or
+ * cache).
+ */
+public class WanEventCounter {
+    private static final ConstructorFunction<String, EventCounter> EVENT_COUNTER_CONSTRUCTOR_FN
+            = new ConstructorFunction<String, EventCounter>() {
+        @Override
+        public EventCounter createNew(String ignored) {
+            return new EventCounter();
+        }
+    };
+    private final ConcurrentHashMap<String, EventCounter> eventCounterMap = new ConcurrentHashMap<String, EventCounter>();
```

Comment:
Do we need to remove related `EventCounter` from `eventCounterMap` upon map destroy?

Reply:
Hm, might make sense 👍

Reply:
Done.